integration-cli/docker_cli_run_unix_test.go

```diff
@@ -619,7 +619,10 @@ func (s *DockerSuite) TestRunOOMExitCode(c *check.C) {
 	errChan := make(chan error)
 	go func() {
 		defer close(errChan)
-		out, exitCode, _ := dockerCmdWithError("run", "-m", "4MB", "busybox", "sh", "-c", "x=a; while true; do x=$x$x$x$x; done")
+		// the original test fails on rhel_7.4, rhel_7.3-dm and rhel_7.4-dm
+		// memory limit lower than 7MB will very likely result in a error of "device or resource busy"
```

Comment:
do we know why?

/cc @kolyshkin

Reply:
The reason _might_ be that the current usage is above the value we're trying to use as a limit, and as kernel can't guarantee this limit to be enforced, it replies with EBUSY.  I have yet to check that though, and don't have an idea why it works with other kernels.

Reply:
I was able to repro this on my laptop (Ubuntu 18.10, vanilla kernel 4.16.12) so it's not specific to RHEL:
```
kir@kd:~/git/rpm$ time docker run -m 4MB busybox sh -c 'x=aaaa; while true; do x=$x$x$x$x; done'; echo $?
docker: Error response from daemon: OCI runtime create failed: container_linux.go:296: starting container process caused "process_linux.go:398: container init caused \"process_linux.go:365: setting cgroup config for procHooks process caused \\\"failed to write 4194304 to memory.limit_in_bytes: write /sys/fs/cgroup/memory/docker/c766d5121094498b1d88a37ebddf67aa7379b177cd9eb38efa9bcc4aa75aa842/memory.limit_in_bytes: device or resource busy\\\"\"": unknown.
ERRO[0000] error waiting for container: context canceled 

real	0m0.732s
user	0m0.011s
sys	0m0.003s
125
```

Reply:
Hmm, I was correct earlier; kernel code tries to set the limit, if the limit can't be set because current usage is more than the new limit, or because kernel can't free any pages it returns EBUSY (meaning the limit is too low). Code is mm/memcontrol.c, function mem_cgroup_resize_limit.

So
(1) this is not kernel version specific (although recent kernels appear to try harder to release memory to set the limit)
(2) this is correct kernel behavior, not a bug
(3) it seems that docker needs more than 4MB to start a container.

I'd suggest to
1. remove the comment about rhel 7.4 et al kernels replacing it with something like

> it appears that 8MB of memory is the minimum needed to reliably start a container

2. raise the limit to 8MB